src/tasks/TaskHelper.ts

```diff
@@ -0,0 +1,166 @@
+    let buildTasks = await tasks.fetchTasks({ type: 'docker-build' });
+    buildTasks =
+        buildTasks.filter(t => t.execution.args.some(a => { // Find all build tasks where an argument to 'docker build' is this Dockerfile
+            let arg: string;
+            if (typeof a === 'string') {
+                arg = a;
+            } else {
+                arg = a.value;
+            }
+
+            arg = resolveWorkspaceFolderPath(folder, arg);
+            return arg === dockerfile;
+        }));
+
+    if (buildTasks.length === 1) {
+        return buildTasks[0]; // If there's only one build task, take it
+    } else if (buildTasks.length > 1) {
+        return buildTasks.find(t => t.name === 'docker-build: release') || buildTasks[0]; // If there's multiple try finding one with the name 'docker-build: release', else take first
+    }
+
+    return undefined;
+}
+
+export function resolveWorkspaceFolderPath(folder: WorkspaceFolder, folderPath: string): string {
+    return folderPath.replace(/\$\{workspaceFolder\}/gi, folder.uri.fsPath);
+}
+
+export function unresolveWorkspaceFolderPath(folder: WorkspaceFolder, folderPath: string): string {
+    // tslint:disable-next-line: no-invalid-template-strings
+    return folderPath.replace(folder.uri.fsPath, '${workspaceFolder}').replace(/\\/g, '/');
+}
+
+// tslint:disable-next-line: no-any
+async function recursiveFindTaskByType(allTasks: TaskDefinition[], type: string, node: any): Promise<TaskDefinition | undefined> {
+    if (!node) {
+        return undefined;
+    }
+
+    // tslint:disable: no-unsafe-any
+    if (node.preLaunchTask) { // node is a debug configuration
+        const next = await findTaskByLabel(allTasks, node.preLaunchTask);
+        return await recursiveFindTaskByType(allTasks, type, next);
+    } else if (node.type === type) { // node is the task we want
+        return node;
+    } else if (node.dependsOn) { // node is another task
+        if (Array.isArray(node.dependsOn)) {
+            for (const label of node.dependsOn as string[]) {
+                let next = await findTaskByLabel(allTasks, label);
+                next = await recursiveFindTaskByType(allTasks, type, next);
+
+                if (next) {
+                    return next;
+                }
+            }
+
+            return undefined;
+        } else {
+            const nextType = node.dependsOn.type;
+            const next = await findTaskByType(allTasks, nextType);
+            return await recursiveFindTaskByType(allTasks, type, next);
+        }
+    }
+    // tslint:enable: no-unsafe-any
+
+    return undefined;
+}
+
+async function findTaskByLabel(allTasks: TaskDefinition[], label: string): Promise<TaskDefinition | undefined> {
+    return allTasks.find(t => t.label === label);
+}
+
+async function findTaskByType(allTasks: TaskDefinition[], type: string): Promise<TaskDefinition | undefined> {
+    return allTasks.find(t => t.type === type);

```

Comment:
>find [](start = 20, length = 4)

Intentionally case-sensitive?

Reply:
Yes, VSCode is case-sensitive about task type.